yamux/src/connection.rs

```diff
@@ -738,30 +721,28 @@ impl<T: AsyncRead + AsyncWrite + Unpin> Active<T> {
                 if is_finish {
                     shared.update_state(self.id, stream_id, State::RecvClosed);
                 }
-                shared.window = shared.window.saturating_sub(parsed_frame.body_len());
-                shared.buffer.push(frame.into_buffer(), HEADER_SIZE);
```

Comment:
I introduced the second parameter to avoid the copy of the body. Do you really want to revert this optimization ? I did notice you suggest to use BytesMut in order to split of the body, however I'm not sure its a good idea to do it in this PR.

Reply:
I actually re-did the entire patch-set to reduce as much diff as possible. Compared to `master`, there is no regression here :)

I'd suggest we do one thing at a time and focus on optimising how we write frames with this PR. Reducing allocations can be left to a later point!